xarray/coding/cftimeindex.py

```diff
@@ -259,6 +259,36 @@ def __new__(cls, data, name=None):
         result._cache = {}
         return result
 
+    @property
+    def data(self):
+        return self.values
+
+    def __repr__(self):
+        """
+        Return a string representation for this object.
+
+        copied from pandas.io.printing.py
+        expect for attrs.append(("calendar", self.calendar))
+        """
+        klass_name = type(self).__name__
+        data = self._format_data()
+        attrs = self._format_attrs()
```

Comment:
I just copied those from pandas...

Reply:
Yes, sorry, I understand.  The issue is that, as private methods, pandas could remove or rename them at any moment.  Note that currently in CFTimeIndex we do not call any private methods that are not defined in the CFTimeIndex class itself.

Within reason, a way to work around this is to copy the private methods over into CFTimeIndex from pandas, but sometimes these private methods call other private methods etc., and things can get out of hand.  I'll look into what it might take and try to make a decision on whether it would be possible to port the pandas repr over, or if we should write our own repr, which might not look exactly like pandas's, but would get the job done.

Reply:
I looked a bit into pd.Index:
_format_data calls format_object_summary in https://github.com/pandas-dev/pandas/blob/bf613c14f41b624b432d0d6b9a29007e7990d460/pandas/io/formats/printing.py. seem quite a loot to rebuild. On the other hand I understand we dont want to depend on these methods. I would rather argue that we only want a string returned and _format_data and _format_attrs sound very standard, so hopefully they would remain. But I am really not an expert on this, just my naive thoughts.

Reply:
Indeed, it does look quite involved in pandas!  Let's maybe explore what we can leverage from xarray a little bit.

Following @dcherian's suggestion, I looked into xarray's machinery for reprs, and I think we could fairly straightforwardly use [`formatting.format_array_flat`](https://github.com/pydata/xarray/blob/2ba530026fb273a2882869a6e09ede053a0f081b/xarray/core/formatting.py#L167-L216) to create a repr that looked like this:
```
CFTimeIndex([2000-01-01 00:00:00 2000-01-02 00:00:00 ... 2000-01-10 00:00:00],
            length=10, name='foo', calendar='gregorian', dtype='object')
```
For context, `format_array_flat` is what is used to generate the short summary of the data for each variable in the Dataset repr, or the variables in the coordinates section in the DataArray repr.  

The downside is that it only shows a maximum of three times from the index for the standard display width (i.e. it won't extend to show values on multiple lines).  Also, unlike the pandas repr, the values are not separated by commas.  If we wanted, I think we could discuss how we might adapt what's in `formatting.py` to address one or both of those issues, or we could go with something that looks like the above for now.  I'm open to either!

Reply:
implemented the quick version for now. I am also open to implement commata and multiple lines.